includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -670,7 +670,7 @@ public function should_use_stripe_platform_on_checkout_page() {
 		if (
 			WC_Payments_Features::is_woopay_eligible() &&
 			'yes' === $this->get_option( 'platform_checkout', 'no' ) &&
-			! ( WC_Payments_Features::is_upe_legacy_enabled() && ! WC_Payments_Features::is_upe_split_enabled() && ! WC_Payments_Features::is_upe_deferred_intent_enabled() ) &&
+			! ( WC_Payments_Features::is_upe_legacy_enabled() && ! WC_Payments_Features::is_upe_split_enabled() ) &&
```

Comment:
```suggestion
			! ( WC_Payments_Features::is_upe_legacy_enabled() && ! ( WC_Payments_Features::is_upe_split_enabled() || WC_Payments_Features::is_upe_deferred_intent_enabled() ) ) &&
```

Shouldn't this be something like the above? So that if legacy UPE is enabled and either UPE split or UPE deferred intent is also enabled--both of which now support WooPay--we can use the Stripe platform account at checkout, if desired. Otherwise if I have the legacy UPE flag enabled, the split UPE flag disabled, but the deferred intent flag enabled, this will return `false` and WooPay will be disabled--even with the deferred intent UPE enabled.

This could be the case when we eventually migrate legacy UPE users to the deferred intent UPE, as they'll probably skip being migrated to the original split UPE. 🤷

Reply:
Yes, you're right! Fixed in https://github.com/Automattic/woocommerce-payments/pull/6656/commits/8991d36285253931f956bb16c970c8063693994f. This `if` statement shouldn't be so complex, as it's challenging to understand. I think things will improve after split UPE/legacy UPE cleanup, but as of now I encapsulated this logic in a separate function (https://github.com/Automattic/woocommerce-payments/pull/6656/commits/8991d36285253931f956bb16c970c8063693994f), so that it's a bit easier to read.